src/PackageManagement/NuGetPackageManager.cs

```diff
@@ -382,6 +385,7 @@ private void InitializePackagesFolderInfo(string packagesFolderPath)
                     var action = NuGetProjectAction.CreateInstallProjectAction(updateToIdentity, primarySources.FirstOrDefault());
 
                     var lowLevelActions = new List<NuGetProjectAction>();
+                    lowLevelActions.Add(NuGetProjectAction.CreateUninstallProjectAction(installedPackageReference.PackageIdentity));
```

Comment:
Build integrated/DNX do not have a reinstall concept, every restore is sort of a reinstall